pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/WocMetric.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.metrics.impl;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration.TypeKind;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaOperationSigMask;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaOperationSignature.Role;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaSignature.Visibility;
+import net.sourceforge.pmd.lang.metrics.MetricOptions;
+
+/**
+ * Weight of class.
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class WocMetric extends AbstractJavaClassMetric {
+
+    @Override
+    public boolean supports(ASTAnyTypeDeclaration node) {
+        return node.getTypeKind() == TypeKind.CLASS;
+    }
+
+
+    @Override
+    public double computeFor(ASTAnyTypeDeclaration node, MetricOptions options) {
+
+        JavaOperationSigMask mask = new JavaOperationSigMask();
+        mask.restrictRolesTo(Role.METHOD, Role.STATIC);
```

Comment:
It doesn't select only static methods, it selects static methods *and* "normal" methods ^^ Maybe it's unclear, it's essentially the same as writing `mask.forbid(Role.CONSTRUCTOR, Role.GETTER_OR_SETTER)`.